datafusion/expr/src/utils.rs

```diff
@@ -260,7 +252,11 @@ pub fn grouping_set_to_exprlist(group_expr: &[Expr]) -> Result<Vec<&Expr>> {
         }
         Ok(grouping_set.distinct_expr())
     } else {
-        Ok(group_expr.iter().collect())
+        Ok(group_expr
+            .iter()
```

Comment:
wondering if can get rid of double collection iteration?

Reply:
The double iteration is used to deduplicate the values